Added better error messages in Flex with yylineno + some minor visual fixes

Fiwo735 opened the pull request:
Addressing #27

dwRchyngqxs commented:
I think it's a improvemet, but we can do even better:
- to avoid confusion, the error function should be yeeted out of the lexer as `yyerror` is only called by the parser,
- bison seems to be able to provide more precise error description according to [this documentation page](https://www.gnu.org/software/bison/manual/html_node/Error-Reporting-Function.html),
- and the lexer ignores all the bad characters (see the last lexer rule before the definition of `yyerror`) which may be fine, but producing a warning in stderr would be better for diagnostic purposes imo.
Hope I'm not asking too much with this. To be fair I think we can merge as is, but some of the improvements don't event take that much time. If you don't want to look into it I'll open a new issue and try myself.

Fiwo735 commented:
Thanks for the feedback.

- Moved `yyerror` to parser
- Used Bison's detailed errors - they are better than the previous solution as they also tell what token was expected. I didn't manage to get the `%locations` and `%define api.pure full` stuff to work, but I reckon the current approach is good enough as it already says "near" what token the error occured, instead of the column position in line that `%locations` seems to offer.
- Added a warning in the last lexer rule (and `UNKNOWN` token for that), although I didn't manage to trigger it for some reason - the detailed error handling from the parser showed up instead.

Let me know if you think any further improvements could be made.

dwRchyngqxs commented:
After adding `$` in the example test I got the following message:
```
Unknown token: $
Error: syntax error, unexpected UNKNOWN, expecting INT_CONSTANT or ';' at line 3 near '$'
```
So it works.

I looked at column tracking and it seems to be much more of a pain so we can give up on that.

Thank you for solving this issue.
